src/jpgimage.cpp

```diff
@@ -179,21 +182,24 @@ DataBuf Photoshop::setIptcIrb(const byte* pPsData, size_t sizePsData, const Iptc
     if (rawIptc.size() & 1)
       psBlob.push_back(0x00);
   }
-  // Write existing stuff after record,
-  // skip the current and all remaining IPTC blocks
+
+  // Write existing stuff after record, skip the current and all remaining IPTC blocks
   size_t pos = sizeFront;
-  while (0 == Photoshop::locateIptcIrb(pPsData + pos, sizePsData - pos, &record, &sizeHdr, &sizeIptc)) {
+  long nextSizeData = Safe::add<long>(static_cast<long>(sizePsData), -static_cast<long>(pos));
+  enforce(nextSizeData >= 0, ErrorCode::kerCorruptedMetadata);
```

Comment:
The fix is in lines 188-189 & 196-197. I ended up using `Safe::add<long>` to make sure that there is not integer overflow and I also used later `enforce` to make sure that the substraction is positive.